include/wasm/WasmModule.h

```diff
@@ -100,23 +101,10 @@ class WasmModule
 
     virtual size_t getMemorySizeBytes();
 
-    // ----- CoW memory -----
-    virtual void writeMemoryToFd(int fd);
-
-    virtual void mapMemoryFromFd();
-
     // ----- Snapshot/ restore -----
-    void snapshotToFile(const std::string& filePath);
-
-    std::vector<uint8_t> snapshotToMemory();
-
-    size_t snapshotToState(const std::string& stateKey);
+    std::string snapshot();
 
-    void restoreFromFile(const std::string& filePath);
-
-    void restoreFromMemory(const std::vector<uint8_t>& data);
-
-    void restoreFromState(const std::string& stateKey, size_t stateSize);
+    void restore(const std::string& snapshotKey);
```

Comment:
Previously there were three forms of snapshot and restore, to files, memory and state. Now we just have the one (to memory)